hilo.rb

```diff
@@ -1,34 +1,113 @@
 #!/usr/bin/env ruby
-puts "Welcome to the Hi-Lo game.\nGuess a random whole number from 1 to 100 (inclusive)."
-puts "You will have a total of 6 guesses."
-puts ""
 
-random_num = rand(1..100)
+# Game to guess random number
+class HiLoGame
 
-# gives the user 6 guesses
-for i in 0..5
-    guesses_left = 6 - i
-    puts "You have #{guesses_left} guesses left."
-    puts "Please guess a whole number."
+    # Constructor to initialise the class with a maximum number of guesses
+    def initialize(max_guesses)
+	@max_guesses = max_guesses
+	@guesses_left = @max_guesses
+	@already_guessed = Array.new
+	@random_num = get_random_num
+	welcome
+    end
+
+    def get_random_num
+	random_num = rand(1..100)
+    end
+
+    # Welcome message for the game
+    def welcome
+	puts "Welcome to the Hi-Lo game.\nGuess a random whole number from 1 to 100 (inclusive)."
+	puts "You will have a total of #{@max_guesses} guesses.\nYou cannot guess the same number more than once."
+	puts ""
+    end
+
+    # The game is started
+    def play
+
+	# Gives the user max number of guesses
+	while game_running?
+	    print_guesses_left
+	    guess = guess_a_number
+	    check_guess(guess)
+	    @guesses_left -= 1
+
+	    if has_won?(guess)
+	        print_has_won
+	    	return
+	    end
```

Comment:
It seems odd to put this in the loop as it happens after the game is finished?

Reply:
Done refactoring in https://github.com/amykingOW/Hi-Lo-Game/pull/4/commits/338e80c8d26ae156122069be2e9765525c520c55. I haven't changed this part before discussing. My thinking is to check if they have won after each guess as they might win before their last turn. If it's outside the while loop then it won't be checked until all the turns are finished. A workaround to this would be to check if they have won in the check_guess method, however I can't think of a way to then finish the program gracefully without using break or exit. Using return here would just take it back to the while loop which would continue until all turns have been used, defeating the purpose of checking for a win in check_guess.

Reply:
It is possible. The question I'd ask you is "Is the game still running if they've guessed the correct number?"

Reply:
Done in https://github.com/amykingOW/Hi-Lo-Game/pull/4/commits/70169447579e15ce11598b19760c920f798023cc.